elasticapm/context/__init__.py

```diff
@@ -27,3 +27,25 @@
 #  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
 #  OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 #  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+def init_execution_context():
+    # If gevent has monkey patched thread local, we should not be using
+    # contextvars as this implies that greenlets are in use
+    if threading_local_monkey_patched():
+        from elasticapm.context.threadlocal import execution_context
+        return execution_context
+
+    try:
+        from elasticapm.context.contextvars import execution_context
+    except ImportError:
+        from elasticapm.context.threadlocal import execution_context
+    return execution_context
+
+
+def threading_local_monkey_patched():
```

Comment:
While we're at it, it might make sense to do the same check for eventlet

http://eventlet.net/doc/patching.html#eventlet.patcher.is_monkey_patched